### PR TITLE
2641 incorrect module names in psydata

### LINKED ISSUE #2641: Incorrect module and region names
In examples/gocean/eg5/profile, the output atm is:
```
 PreStart called for module 'invoke_0' region 'r0'
 PostEnd called for module 'invoke_0' region 'r0'
 PreStart called for module 'invoke_1_update_field' region 'r0'
 PostEnd called for module 'invoke_1_update_field' region 'r0'
```
while originally it was:
```
 PreStart called for module 'psy_test' region 'invoke_0:r0'
 PostEnd called for module 'psy_test' region 'invoke_0:r0'
 PreStart called for module 'psy_test' region 'invoke_1_update_field:update_field_code:r0'
 PostEnd called for module 'psy_test' region 'invoke_1_update_field:update_field_code:r0'
```
I am not sure where/when the module name was changed, I believe it should be `psy_test` as it was originally (since this is indeed the module).

Investigate why/when this was changed, and fix this issue.

Optionally: improve testing to check this output

### PR DESCRIPTION
**Pull request by hiker**
Fixes #2641.

### CODE REVIEW

File: src/psyclone/psyir/nodes/psy_data_node.py [hunk 1]
```diff
@@ -756,10 +757,16 @@ def gen_type_bound_call(typename, methodname, argument_list=None,
                 f"lowering but '{self}' is not.")
 
         self.generate_symbols(routine_schedule.symbol_table)
-
         module_name = self._module_name
         if module_name is None:
-            module_name = routine_schedule.name
+            container = routine_schedule.ancestor(Container)
+            # See if there current code is inside of a module. If so,
```
Comment (arporter) [inline src/psyclone/psyir/nodes/psy_data_node.py:763]: "If the current code is inside a module use the module name, otherwise ...."

**Review by arporter — CHANGES_REQUESTED** [review]
Looks good Joerg. Thanks for tracking this down.
You'll need to update the tutorials too - at least `tutorial/practicals/nemo/2_nemo_profiling` needs tweaking.
Does the User/Dev Guide have examples that need updating too?

File: examples/gocean/eg5/profile/test.x90 [hunk 2]
```diff
@@ -37,10 +37,10 @@
 !> stdout. Expected output:
 !>
 !>  profile_PSyDataInit called
-!>  PreStart called for module 'psy_test' region 'invoke_0:r0'
-!>  PostEnd called for module 'psy_test' region 'invoke_0:r0'
-!>  PreStart called for module 'psy_test' region 'invoke_1_update_field:update_field_code:r0'
-!>  PostEnd called for module 'psy_test' region 'invoke_1_update_field:update_field_code:r0'
+!>  PreStart called for module 'psy_test' region 'invoke_0-r0'
+!>  PostEnd called for module 'psy_test' region 'invoke_0-r0'
+!>  PreStart called for module 'psy_test' region 'invoke_1_update_field:update_field_code-r0'
```
[the remark this answers is not in the document]
Comment (hiker) [reply examples/gocean/eg5/profile/test.x90]: They should indeed be.

File: src/psyclone/psyir/nodes/psy_data_node.py (hunk 1 above)
Comment (hiker) [reply src/psyclone/psyir/nodes/psy_data_node.py]: Done.

**Comment by hiker** [comment]
> Looks good Joerg. Thanks for tracking this down. You'll need to update the tutorials too - at least `tutorial/practicals/nemo/2_nemo_profiling` needs tweaking. 

Done, indeed I missed the example in the Fortran file (maybe I should just have removed that :) ).

> Does the User/Dev Guide have examples that need updating too?

Yes, it did - a bit strange, I was certain I had updated that section, but maybe I am mixing things up. I've updated the user guide, added examples for names used in transformation, and shortened the example for DSL naming (and properly indented them).

**Review by arporter — APPROVED** [review]
All requested changes have been made, thanks. (I too got a sense of deja vu when looking at the UG changes but I think that's because your last PR also changed things there?)
Updated UG builds fine and looks good.
Will proceed to merge.

**Pull request merged**